### ISSUE TITLE
Red Faction® II [54510005]

### ISSUE THREAD

**Issue by CakeLancelot** [Cxbx-Reloaded/game-compatibility#410]
### Description

Shows intros (though they are too fast even with vsync), shows loading screen (textures are garbled), then crashes.

```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x001B41CB(=CDirectSound_CommitDeferredSettings+0xee)

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Cancel   
---------------------------

```

### Dumps 

[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1367081/KrnlDebug.txt)
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1367082/Xbe.txt)

### Status
**Previous behavior:** 

None reported

**Current behavior:**

Shows intros and loading screen, crashes.

### Screenshots 
![cxbx_2017-10-08_12-56-01](https://user-images.githubusercontent.com/5341136/31319421-ce8d3c04-ac28-11e7-8502-48cf7bed841c.png)
![cxbx_2017-10-08_12-56-04](https://user-images.githubusercontent.com/5341136/31319423-ce8e8988-ac28-11e7-8cab-1f32afd9be98.png)
![cxbx_2017-10-08_12-56-18](https://user-images.githubusercontent.com/5341136/31319422-ce8e16f6-ac28-11e7-99b1-7198871da4d0.png)

[Video](https://www.youtube.com/watch?v=c3SZ4oykw-A)

### Additional Information

Tested on version 94a2ddae (Oct  8 2017)

**Comment by PatrickvL** [comment]
Looks like palettized textures not getting converted. Should be interesting to see how this behaves in my WIP_LessVertexPatching branch, which supports converting near all texture formats.

**Comment by CakeLancelot** [comment]
> Looks like palettized textures not getting converted. Should be interesting to see how this behaves in my WIP_LessVertexPatching branch, which supports converting near all texture formats.

Yep. I posted it on Discord previously but I figured I should probably post here as well: The loading screen is *much* improved on the WIP_LessVertexPatching branch.

![cxbx_2017-10-13_15-50-35](https://user-images.githubusercontent.com/5341136/31655864-d39f8198-b2ef-11e7-962d-f055a45f1a36.png)
(Not pictured: An additional splash screen also appears. Intro videos are now no longer visible, but you can hear them playing.)

**Comment by CakeLancelot** [comment]
After the HLEDB_v2 merge this title suffers a regression - it crashes instantly whereas before it would display intro videos and a loading screen. Here's the HLE cache file before and after (needed to be zipped as github doesn't support ini files) - I've also attached a log of the game running on the latest commit - a62d73c1 (Oct 20 2017).

Error Message:
```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x77D5DDE3

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Cancel   
---------------------------
```
[Red Faction II Caches.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/1403262/Red.Faction.II.Caches.zip)
[KrnlDebug2.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1403263/KrnlDebug2.txt)

**Comment by RadWolfie** [comment]
I took a looked at before and after cache file @CakeLancelot provided. Major introduced is XACT library detection. Have you tried remove all of XACT and IXACT from current cache file? I think title will make further progress. Except audio might not be working properly due to certain LL values are setup separately.

**Comment by CakeLancelot** [comment]
> Have you tried remove all of XACT and IXACT from current cache file? I think title will make further progress.

I've tried this and just as you described, the title now goes into menus and ingame. There are still the aforementioned issues - and it seems it wasn't just the intro videos that were too fast, the entire game is sped up a lot even with vsync on. Both singleplayer and multiplayer work but singleplayer takes a *long* time to load (not sure if that's the case on actual hardware yet). Here's some screenshots and a video:

![cxbx_2017-10-21_17-14-14](https://user-images.githubusercontent.com/5341136/31856377-59658e06-b685-11e7-9a3a-d67c21fd506d.png)
![cxbx_2017-10-21_17-17-27](https://user-images.githubusercontent.com/5341136/31856378-597f3fea-b685-11e7-9e3c-3a2741ba1efa.png)
![cxbx_2017-10-21_17-17-45](https://user-images.githubusercontent.com/5341136/31856380-59ac7a8c-b685-11e7-886b-5b0561535ff9.png)
![cxbx_2017-10-21_17-18-13](https://user-images.githubusercontent.com/5341136/31856381-59e73ab4-b685-11e7-976b-fafde52495ca.png)
![cxbx_2017-10-21_17-37-20](https://user-images.githubusercontent.com/5341136/31856440-d28c39c8-b686-11e7-8b91-300cf41a293f.png)
[Video](https://youtu.be/G0WiV4a-tRY)

**Comment by CakeLancelot** [comment]
Big improvements for this title recently, only major issue now is the speed - the game is way too fast to be playable:
![cxbx_2018-03-04_15-57-07](https://user-images.githubusercontent.com/5341136/36952359-7ed7f1ac-1fd4-11e8-975d-f513d426a855.png)
![cxbx_2018-03-04_15-58-53](https://user-images.githubusercontent.com/5341136/36952360-7ef4c7b4-1fd4-11e8-846d-1095ccf5551f.png)
![cxbx_2018-03-04_17-44-13](https://user-images.githubusercontent.com/5341136/36952361-7f59deec-1fd4-11e8-9766-18f874d8ae07.png)
![cxbx_2018-03-04_17-44-20](https://user-images.githubusercontent.com/5341136/36952362-7f7781e0-1fd4-11e8-9871-821af058fdcf.png)

**Comment by ObiKKa** [comment]
Just a reminder that the timing in this game is now improved. Also, a black square occasionally covers the top-left corner of the screen depending on your location and orientation.
- [Cxbx-Reloaded Xbox Emulator - Red Faction II Ingame! (3c201d7 + WIP / HW Mode, FXAA)](https://www.youtube.com/watch?v=kuKMhsL_Oyg&t=255s&index=36&list=PL4PD8cV5ypNDPTJbrol7A-5rvuteHWWYj) (John GodGames Emus. Published on Mar 6, 2018. Build came just before Luke's vblank timing fix, though has his QueryPerformanceCounter fix; but the timing here is good.)

**Comment by PatrickvL** [comment]
Radwolfie stated
> I can verified Red Faction 2 only show the black square if an enemy is in front of you with white and black linear texture pattern on the floor  (which should be a shadow). Once they're dead and gone or look away from them. It disappeared.

**Comment by RadWolfie** [comment]
By black square, I'm talking where you can see it at top-left corner. Just to be clear for other to understand.
